pkg/autodetect/main.go

```diff
@@ -31,6 +32,7 @@ var _ AutoDetect = (*autoDetect)(nil)
 type AutoDetect interface {
 	Platform() (platform.Platform, error)
```

Comment:
Now what is the difference in the implementation between `Platform` and `OpenShiftRoutesAvailability`? 

The platform used to check the route presence.

Reply:
From #1418
> The detection checks if OpenShift Routes API is installed. If true, it means we're running the Operator in OpenShift. The problem is that, for the E2E tests, we're installing the OpenShift Routes in KiND (Kubernetes). So, the E2E tests think they are running in OpenShift.

We can need to differentiate when we are really running in OpenShift and when not. As for things like #1338

Reply:
Is the Platform then still used for something else? Like TLS settings or other things?

Reply:
I will use it to fix #1338